### PR TITLE
fix: Updated the regex in Applied Visual Design: Adjust the background-color Property of Text CSS test

### LINKED ISSUE #39823: Applied Visual Design: Adjust the background-color Property of Text CSS test too strict
This lesson: https://www.freecodecamp.org/learn/responsive-web-design/applied-visual-design/adjust-the-background-color-property-of-text

Requires the semi-colon at the end of a last declaration in the block.
Passes
```
h4 {
  text-align: center;
  padding: 10px;
  background-color: rgba(45, 45, 45, 0.1);
}
```
Fails:
```
h4 {
  text-align: center;
  padding: 10px;
  background-color: rgba(45, 45, 45, 0.1)
}
```

---

The test should be changed to allow the missing semi-colon, provided the `background-color` declaration is last in the scope.

Similar implementation to this should be followed: https://github.com/freeCodeCamp/freeCodeCamp/pull/39570/files

### PR DESCRIPTION
**Pull request by OllieClark33**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #39823

As mentioned by @Skye020 in issue #39823, I have made the semi-colon after the background-colour declaration optional, but only if it is the last declaration in the scope.

### CODE REVIEW

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-background-color-property-of-text.md
Comment (nhcarrigan) [inline curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-background-color-property-of-text.md:29]:
```suggestion
    testString: assert(code.match(/(background-color|background):\s*?rgba\(\s*?45\s*?,\s*?45\s*?,\s*?45\s*?,\s*?0?\.1\s*?\);?/gi));
```

This seems a bit more DRY and still worked locally. Is there a particular reason for checking a whitespace after the semicolon that I might be missing?

**Review by naomi-lgbt — DISMISSED** [review]

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-background-color-property-of-text.md
Comment (Sky020) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-background-color-property-of-text.md:29]: This will cause the tests to pass on false positives. The check must be made that `;` is only optional if the next object is a `}`

Comment (nhcarrigan) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-background-color-property-of-text.md:29]: Ah - thanks for the clarification! 🙂

**Review by naomi-lgbt — DISMISSED** [review]
In that case, this LGTM~! 🥳 

Congratulations @OllieClark33 on your first pull request and thank you for your interest in contributing! Our workflow will require at least one additional approval before this can be merged.

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-background-color-property-of-text.md
Comment (Sky020) [inline curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-background-color-property-of-text.md:29]:
```suggestion
    testString: assert(/(background-color|background):rgba\(45,45,45,0?\.1\)(;?}|;)/gi.test(code.replace(/\s/g, '')));
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
This is just nicer to look at

File: curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-background-color-property-of-text.md
Comment (OllieClark33) [reply curriculum/challenges/english/01-responsive-web-design/applied-visual-design/adjust-the-background-color-property-of-text.md:29]: I agree! Thanks

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍 Thank you, for making that change @OllieClark33 

@nhcarrigan Merge if you are happy, pending tests. 😃

**Review by naomi-lgbt — APPROVED** [review]
LGTM~! 🥳 

Congratulations on your first PR! We look forward to future contributions from you!

**Pull request merged**
